config.c

```diff
@@ -112,12 +113,16 @@ static RRStatus processConfigParam(const char *keyword, const char *value,
     /* Parameters we don't accept as config set */
     if (!on_init && (!strcmp(keyword, CONF_ID) ||
                 !strcmp(keyword, CONF_RAFT_LOG_FILENAME) ||
-                !strcmp(keyword, CONF_SLOT_CONFIG) ||
                 !strcmp(keyword, CONF_EXTERNAL_SHARDING))) {
         snprintf(errbuf, errbuflen-1, "'%s' only supported at load time", keyword);
         return RR_ERROR;
     }
 
+    if (redis_raft.state != REDIS_RAFT_UNINITIALIZED && !strcmp(keyword, CONF_SLOT_CONFIG)) {
+        snprintf(errbuf, errbuflen-1, "'%s' only supported before cluster init/join", keyword);
+        return RR_ERROR;
+    }
```

Comment:
Looks like this function doesn't know about redis_raft. Maybe we should add `before_join` parameter similar to `on_init`. We'll pass `state == REDIS_RAFT_UNINITIALIZED` into it. Then, reject slot config if `before_join` is false. That requires initializing shard info after this function(rather than calling `ShardingInfoReset()` here). Probably something like just before setting `state = REDIS_RAFT_UP`, we init sharding info if it is not initialized already. That way we'll have updated config if any.

Reply:
what do you mean by doesn't know?

Reply:
There is no usage of `redis_raft` here in this config.c file. Instead of checking `redis_raft.state != REDIS_RAFT_UNINITIALIZED`, this function accepts `on_init` parameter. Otherwise, it could do that check inside, it wouldn't need that function parameter. 
So, should we follow that convention? Can we avoid doing `redis_raft.state !=  REDIS_RAFT_UNINITIALIZED` and avoid calling `ShardingInfoReset()` here ?